### PR TITLE
chart: Set module.diagnostics.enabled=false by default

### PR DESCRIPTION
**Pull request by LukeShu**
## Description

The main goal of this PR is to take advantage of the chart's major version bump to make a backward-incompatible change: To change the default from `module.diagnostics.enabled=true` to `module.diagnostics.enabled=false`. It has long been our documented best-practice to disable it for production, but to avoid breaking backward-compatibility, we've had to leave it on by default.  The alternative solution (which we should also do, but later) that does not require a major version bump is to change the `Module` CRD's default in a later apiVersion (`getambassador.io/v3alpha2`).  But we're not quite ready for a new apiVersion right now, but we should roll this change out to users ASAP; so do it in the chart and default YAMLs.

But while we're getting there, tighten up `tools/src/filter-yaml` to help catch mistakes.

## Related Issues
- https://github.com/datawire/saas_infra/pull/475

## Testing
I made the `filter-yaml` tool stricter to help test that we don't forget things from `require.yaml`, as I initially had when making this change.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`. - yes I did

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale. - should improve things

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [x] My change is adequately tested. - I'd say so

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently. - no tricks

 - [x] The changes in this PR have been reviewed for security concerns and adherence to security best practices. - turning off diagnostics is a security best practice

### CODE REVIEW

File: manifests/emissary/emissary-crds.yaml.in [hunk 1]
```diff
@@ -3015,6 +3015,8 @@ spec:
           metadata:
             type: object
           spec:
+            description: 'TODO(lukeshu): In v3alpha2, get rid of unnecessary nesting
```
Comment (AliceProxy) [inline manifests/emissary/emissary-crds.yaml.in:3018]: Indeed

File: CHANGELOG.md [hunk 2]
```diff
@@ -92,6 +92,13 @@ it will be removed; but as it won't be user-visible this isn't considered a brea
   variable is now a misnomer, as it no longer configures which xDS API version is used, but it still
   affects what the default protocol used for a `TracingService` that points at Zipkin.
 
+- Change: In the standard published `.yaml` files, now included is a `Moduel` resource that disables
```
Comment (AliceProxy) [inline CHANGELOG.md:95]: `Module`?

**Review by AliceProxy — CHANGES_REQUESTED** [review]
Just a small typo, then LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (LukeShu) [reply CHANGELOG.md]: Fixed.

**Comment by LukeShu** [comment]
I've amended the final commit to fix the typo that Alice pointed out
```patch
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 052a5860e..644bc8ebc 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -92,7 +92,7 @@ it will be removed; but as it won't be user-visible this isn't considered a brea
   variable is now a misnomer, as it no longer configures which xDS API version is used, but it still
   affects what the default protocol used for a `TracingService` that points at Zipkin.
 
-- Change: In the standard published `.yaml` files, now included is a `Moduel` resource that disables
+- Change: In the standard published `.yaml` files, now included is a `Module` resource that disables
   the `/ambassador/v0/` → `127.0.0.1:8878` synthetic mapping.  We have long recommended to turn
   this off for production use; it is now off in the standard YAML.  The associated Helm chart
   release also now disables it by default.  A later apiVersion (`getambassador.io/v3alpha2` or
diff --git a/docs/releaseNotes.yml b/docs/releaseNotes.yml
index 0a454ada3..bae0d4b98 100644
--- a/docs/releaseNotes.yml
+++ b/docs/releaseNotes.yml
@@ -57,7 +57,7 @@ items:
       - title: Default YAML disables the diagnostics interface
         type: change
         body: >-
-          In the standard published <code>.yaml</code> files, now included is a <code>Moduel</code>
+          In the standard published <code>.yaml</code> files, now included is a <code>Module</code>
           resource that disables the <code>/ambassador/v0/</code> → <code>127.0.0.1:8878</code>
           synthetic mapping.  We have long recommended to turn this off for production use; it is
           now off in the standard YAML.  The associated Helm chart release also now disables it by
```

**Review by AliceProxy — APPROVED** [review]

**Review by LanceEa — APPROVED** [review]
lgtm

**Pull request merged**